addons/purchase/tests/test_accrued_purchase_orders.py

```diff
@@ -180,3 +180,32 @@ def test_accrued_order_returned(self):
             {'account_id': self.alt_exp_account.id, 'debit': 0.0, 'credit': 1000.0},
             {'account_id': self.account_revenue.id, 'debit': 6000.0, 'credit': 0.0},
         ])
+    
+    def test_error_when_different_currencies_accrued(self):
+        """
+        Tests that if two Purchase Orders with different currencies are selected for Accrued Expense Entry, 
+        a UserError is raised.
+        """
+        purchase_orders = self.env['purchase.order'].create([
+            {
+                'partner_id': self.partner_a.id,
+                'currency_id': self.company_data['currency'].id,
+            }, 
+            {
+                'partner_id': self.partner_a.id,
+                'currency_id': self.currency_data['currency'].id,
+            }
+        ])
+        accrued_account = self.env['account.account'].create({
+            'name': 'Accrued Expenses',
+            'code': '9999',
+        })
+        purchase_orders.button_confirm()
+        accrued_wizard = self.env['account.accrued.orders.wizard'].with_context(
+            active_model='purchase.order',
+            active_ids=purchase_orders.ids,
+        ).create({
```

Comment:
You can use `new()`. You would not need your new account then (and it reflects better what should raise)